api/src/test/java/jakarta/data/metamodel/TextAttributeTest.java

```diff
@@ -32,127 +34,106 @@ static class Author {
     void shouldCreateContainsRestriction() {
-        @SuppressWarnings("unchecked")
-        BasicRestriction<Author,String> restriction =
-                (BasicRestriction<Author, String>) testAttribute.contains("testValue");
-
-        Like like = (Like) restriction.constraint();
-        StringLiteral<?> literal = ((StringLiteral<?>) like.pattern());
+        var restriction = (BasicRestriction<?, ?>) _Author.testAttribute.contains("testValue");
 
         SoftAssertions.assertSoftly(soft -> {
-            soft.assertThat(restriction.expression()).isEqualTo(testAttribute);
-            soft.assertThat(literal.value()).isEqualTo("%testValue%");
+            soft.assertThat(restriction.constraint()).isInstanceOf(Like.class);
+            soft.assertThat(restriction.expression()).isEqualTo(_Author.testAttribute);
+            soft.assertThat(restriction.constraint()).isEqualTo(Like.substring("testValue"));
         });
     }
 
     @Test
+    @DisplayName("should create startsWith restriction")
     void shouldCreateStartsWithRestriction() {
-        @SuppressWarnings("unchecked")
-        BasicRestriction<Author,String> restriction =
-                (BasicRestriction<Author, String>) testAttribute.startsWith("testValue");
-
-        Like like = (Like) restriction.constraint();
-        StringLiteral<?> literal = ((StringLiteral<?>) like.pattern());
+        var restriction = (BasicRestriction<?, ?>) _Author.testAttribute.startsWith("testValue");
 
         SoftAssertions.assertSoftly(soft -> {
-            soft.assertThat(restriction.expression()).isEqualTo(testAttribute);
-            soft.assertThat(literal.value()).isEqualTo("testValue%");
+            soft.assertThat(restriction.constraint()).isInstanceOf(Like.class);
+            soft.assertThat(restriction.expression()).isEqualTo(_Author.testAttribute);
+            soft.assertThat(restriction.constraint()).isEqualTo(Like.prefix("testValue"));
         });
     }
 
     @Test
+    @DisplayName("should create endsWith restriction")
     void shouldCreateEndsWithRestriction() {
-        @SuppressWarnings("unchecked")
-        BasicRestriction<Author,String> restriction =
-                (BasicRestriction<Author,String>) testAttribute.endsWith("testValue");
-
-        Like like = (Like) restriction.constraint();
-        StringLiteral<?> literal = ((StringLiteral<?>) like.pattern());
+        var restriction = (BasicRestriction<?, ?>) _Author.testAttribute.endsWith("testValue");
 
         SoftAssertions.assertSoftly(soft -> {
-            soft.assertThat(restriction.expression()).isEqualTo(testAttribute);
-            soft.assertThat(literal.value()).isEqualTo("%testValue");
+            soft.assertThat(restriction.constraint()).isInstanceOf(Like.class);
+            soft.assertThat(restriction.expression()).isEqualTo(_Author.testAttribute);
+            soft.assertThat(restriction.constraint()).isEqualTo(Like.suffix("testValue"));
```

Comment:
I'm not sure I would say that:

    assertThat(restriction.constraint()).isEqualTo(Like.suffix("testValue"))

is exactly the same thing as:

    assertThat(literal.value()).isEqualTo("%testValue")

I guess I would maybe leave both assertions there.

Reply:
I agree. They are definitely not the same assertions.  Losing the assertion on `.value()` means the test will pass even if `Like.suffix` is wrongly implemented -- as long as the wrong implementation returns a constraint that is equal whatever `.endsWith` returned, which is probably `Like.suffix` invoked on the same parameter.